consensus/system_contract/consensus.go

```diff
@@ -114,8 +114,8 @@ func (s *SystemContract) verifyHeader(chain consensus.ChainHeaderReader, header
 	if header.Nonce != (types.BlockNonce{}) {
 		return errInvalidNonce
 	}
-	// Check that the extra-data contains signature
-	if header.Number.Cmp(big.NewInt(0)) != 0 && len(header.BlockSignature) != extraSeal {
+	// Check that the BlockSignature contains signature if block is not requested
+	if header.Number.Cmp(big.NewInt(0)) != 0 && len(header.BlockSignature) != extraSeal && !header.Requested {
```

Comment:
```suggestion
if !header.Requested && header.Number.Cmp(big.NewInt(0)) != 0 && len(header.BlockSignature) != extraSeal {
```

easier to read and no need to do the checks for requested headers as first part of expression evaluates to false and it aborts immediately without executing the other half